src/ll/argument.rs

```diff
@@ -30,42 +29,50 @@ impl<'a> ArgumentIterator<'a> {
         bytes
     }
 
-    /// Fetch a slice of bytes of the given size. Returns `None` if there's not enough data left.
-    pub fn fetch_bytes(&mut self, amt: usize) -> Option<&'a [u8]> {
-        if amt > self.data.len() {
-            return None;
+    /// Fetch a typed argument. Returns `None` if there's not enough data left.
+    pub fn fetch<T: zerocopy::FromBytes>(&mut self) -> Option<&'a T> {
+        match zerocopy::LayoutVerified::<_, T>::new_from_prefix(self.data) {
+            None => {
+                if self.data.as_ptr() as usize & (core::mem::align_of::<T>() - 1) != 0 {
```

Comment:
I think it would be more clear to check `self.data.as_ptr() as usize % core::mem::align_of::<T>() != 0`. Technically, `align_of` doesn't guarantee that it returns a power of 2 -- although I assume it does on all platforms.

Reply:
Done. I think technically speaking it [must be a power of 2](https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment)

> Alignment is measured in bytes, and must be at least 1, and always a power of 2.

but I agree that using `%` is much clearer and will be optimised down to the same thing anyway.